Shared.IntegrationTests/UploadApi/ArchiveMethodsTest.cs

```diff
@@ -156,35 +154,92 @@ public void TestCreateZip()
+            var publicIds = new List<string> { uploadResult.PublicId };
+
+            var video = ApiShared.GetCloudinaryParam(ResourceType.Video);
+            var archiveParams = new ArchiveParams().ResourceType(video).PublicIds(publicIds);
+            var archiveUrl = m_cloudinary.DownloadArchiveUrl(archiveParams);
+
+            Assert.True(CanArchiveFileBeDownloaded(archiveUrl));
+        }
 
-            var urlStr = m_cloudinary.DownloadArchiveUrl(parameters);
+        [Test]
+        public void TestDownloadArchiveUrlForRaw()
+        {
+            var tag = GetMethodTag();
+            var raw = ApiShared.GetCloudinaryParam(ResourceType.Raw);
+            var uploadParams = new RawUploadParams
+            {
+                File = new FileDescription(m_testPdfPath),
+                Tags = $"{tag},{m_apiTag}"
+            };
+            var uploadResult = m_cloudinary.Upload(uploadParams, raw);
+            var publicIds = new List<string> { uploadResult.PublicId };
 
-            var dicQueryString = new Uri(urlStr).Query.Split('&').ToDictionary(
-                c => Uri.UnescapeDataString(c.Split('=')[0]), c => Uri.UnescapeDataString(c.Split('=')[1])
-            );
+            var archiveParams = new ArchiveParams().ResourceType(raw).PublicIds(publicIds);
+            var archiveUrl = m_cloudinary.DownloadArchiveUrl(archiveParams);
 
-            Assert.AreEqual("download", dicQueryString["mode"]);
-            Assert.AreEqual(archiveTag, dicQueryString["tags[]"]);
+            Assert.True(CanArchiveFileBeDownloaded(archiveUrl));
         }
 
         [Test]
-        public void TestDownloadPrivate()
+        public void TestDownloadZipForImage()
         {
-            long expiresAt = Utils.UnixTimeNowSeconds() + 7200;
+            var tag = GetMethodTag();
+            var uploadParams = new ImageUploadParams
+            {
+                File = new FileDescription(m_testImagePath),
+                Tags = $"{tag},{m_apiTag}"
+            };
+            m_cloudinary.Upload(uploadParams);
 
-            string result = m_cloudinary.DownloadPrivate("zihltjwsyczm700kqj1z", expiresAt: expiresAt);
-            Assert.True(Regex.IsMatch(result, @"https://api\.cloudinary\.com/v1_1/[^/]*/image/download\?api_key=\d*&expires_at=" + expiresAt + @"&public_id=zihltjwsyczm700kqj1z&signature=\w{40}&timestamp=\d{10}"));
+            var archiveUrl = m_cloudinary.DownloadZip(tag, null);
+
+            Assert.True(CanArchiveFileBeDownloaded(archiveUrl));
         }
 
         [Test]
-        public void TestDownloadZip()
+        public void TestDownloadZipForVideo()
         {
-            var result = m_cloudinary.DownloadZip("api_test_custom1", null);
-            Assert.True(Regex.IsMatch(result, @"https://api\.cloudinary\.com/v1_1/[^/]*/image/download_tag\.zip\?api_key=\d*&signature=\w{40}&tag=api_test_custom1&timestamp=\d{10}"));
+            var tag = GetMethodTag();
```

Comment:
Consider reusing `UploadImageForTestArchive()`, and adjust it to support video/raw files as well